runtime/convertValues.go

```diff
@@ -457,6 +469,43 @@ func importPathValue(v cadence.Path) interpreter.PathValue {
 	}
 }
 
+func importTypeValue(
+	_ *interpreter.Interpreter,
+	v cadence.Type,
+) (
+	interpreter.TypeValue,
+	error,
+) {
+	return interpreter.TypeValue{
+		Type: ImportType(v),
+	}, nil
+}
+
+func importCapability(
+	_ *interpreter.Interpreter,
+	path cadence.Path,
+	address cadence.Address,
+	borrowType cadence.Type,
+) (
+	interpreter.CapabilityValue,
+	error,
+) {
+
+	switch borrowType.(type) {
+	case cadence.ReferenceType:
+		return interpreter.CapabilityValue{
+			Path:       importPathValue(path),
+			Address:    interpreter.NewAddressValueFromBytes(address.Bytes()),
+			BorrowType: ImportType(borrowType),
+		}, nil
+	}
+	return interpreter.CapabilityValue{}, fmt.Errorf(
+		"cannot import capability of type '%s', '%s' must be a reference type",
+		borrowType.ID(),
+		borrowType.ID(),
+	)
```

Comment:
Instead of a single-case switch statement, just use a conditional cast.

Also, bail out early instead of nesting the happy path, i.e. invert the logic, e.g.

```suggestion
	_, ok := borrowType.(cadence.ReferenceType)
	if !ok {
		return interpreter.CapabilityValue{}, fmt.Errorf(
			"cannot import capability: expected reference, got '%s'",
			borrowType.ID(),
		)
	}
	return interpreter.CapabilityValue{
		Path:       importPathValue(path),
		Address:    interpreter.NewAddressValueFromBytes(address.Bytes()),
		BorrowType: ImportType(borrowType),
	}, nil
```